cdap-master/src/main/java/co/cask/cdap/master/startup/AbstractMasterCheck.java

```diff
@@ -26,45 +26,61 @@
 /**
  * Base for master startup checks.
  */
-abstract class AbstractMasterCheck extends Check {
+public abstract class AbstractMasterCheck extends Check {
   protected final CConfiguration cConf;
   protected final Set<ServiceResourceKeys> systemServicesResourceKeys;
 
   protected AbstractMasterCheck(CConfiguration cConf) {
     this.cConf = cConf;
+    this.systemServicesResourceKeys = AbstractMasterCheck.createSystemServicesResourceKeysMap(cConf);
+  }
+
+  public static Set<ServiceResourceKeys> createSystemServicesResourceKeysMap(CConfiguration cConf) {
```

Comment:
It is a bad design to just expose method as public static one. Refactor this to a util class (e.g. MasterUtils, a final class with private constructor) to host this method so that it can be used for both the master check and the master main.